riscv/src/lib.rs

```diff
@@ -329,3 +329,62 @@ fn output_files_from_cargo_build_plan(
 
     assemblies
 }
+
+/// Maps an instruction in .insn syntax to Statement::Instruction() in the expected format.
+///
+/// See https://www.rowleydownload.co.uk/arm/documentation/gnu/as/RISC_002dV_002dFormats.html
+pub fn map_insn_i(
+    opcode6: Expression,
+    func3: Expression,
+    rd: Register,
+    rs1: Register,
+    simm12: Expression,
+) -> Statement {
+    let (Expression::Number(opcode6), Expression::Number(func3)) = (opcode6, func3) else {
+        panic!("Only literal opcode and function are supported in .insn syntax");
+    };
+
+    // Except for "csr*i" instructions, these are all instructions in RISC-V
+    // Instruction Set Manual that we are supposed to implement and roughly fits
+    // the pattern the I-type instruction.
+    let name = match (opcode6, func3) {
+        (0b1100111, 0b000) => "jalr",
+        (0b0000011, 0b000) => "lb",
+        (0b0000011, 0b001) => "lh",
+        (0b0000011, 0b010) => "lw",
+        (0b0000011, 0b100) => "lbu",
+        (0b0000011, 0b101) => "lhu",
+        (0b0010011, 0b000) => "addi",
+        (0b0010011, 0b010) => "slti",
+        (0b0010011, 0b011) => "sltiu",
+        (0b0010011, 0b100) => "xori",
+        (0b0010011, 0b110) => "ori",
+        (0b0010011, 0b111) => "andi",
+        (0b0001111, 0b000) => "fence",
+        (0b0001111, 0b001) => "fence.i",
+        (0b1110011, 0b000) => {
+            let Expression::Number(simm12) = simm12 else {
+                panic!("Only literal simm12 is supported for ecall and ebreak instructions");
+            };
+            match simm12 {
+                0 => "ecall",
+                1 => "ebreak",
+                _ => panic!("unknown instruction"),
+            }
+        }
+        (0b1110011, 0b001) => "csrrw",
+        (0b1110011, 0b010) => "csrrs",
+        (0b1110011, 0b011) => "csrrc",
+        // won't interpret "csr*i" instructions because it is too weird to
+        // encode an immediate as a register
+        _ => panic!("unsupported .insn instruction"),
+    };
+
+    let args = vec![
```

Comment:
At least for ecall and ebreak, those should be empty. Maybe also for some others?

Reply:
Yeah... also both fences. Fixed.